core/src/main/java/dev/morphia/query/updates/UpdateOperators.java

```diff
@@ -244,13 +243,13 @@ public static PopOperator pop(String field) {
     /**
      * The $pull operator removes from an existing array all instances of a value or values that match a specified condition.
      *
-     * @param field  the field to update
-     * @param filter the filter to apply
+     * @param field the field to update
+     * @param value the instance of a value or the filter to apply
      * @return the update operator
      * @update.operator $pull
      */
-    public static UpdateOperator pull(String field, Filter filter) {
-        return new PullOperator(field, filter);
+    public static UpdateOperator pull(String field, Object value) {
```

Comment:
technically this break bincompat which I've tried to avoid but I don't think it'll effectively matter and the tools I had in place to check this kind of thing are broken anyway. but something to keep an eye on if users break strangely,.

Reply:
Yea, I thought about that while working on it. If there's a better way to achieve this without breaking compatibility, I'd happily switch to it. 

I thought about adding a new method here, something like `UpdateOperator pullValue(String field, Object value)`. Let me know if something like that is a better approach. Also, is there any concern about changes to `PullOperator`? I see that's marked internal, but not sure what kind of standards there are around that.

Reply:
I mean, we might leave the old `Filter` method have it call the new `Object` method but I'm not sure it's worth it just to satisfy the pedantry of a check that's no longer actively enforced. This is technically a bug fix to the API so if someone complains we can have that conversation then.

Reply:
Makes sense and sounds good to me. Thanks!